WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationAccessUpdateEventProcessor.swift

```diff
@@ -26,15 +27,48 @@ protocol ConversationAccessUpdateEventProcessorProtocol {
     ///
     /// - Parameter event: A conversation access update event.
 
-    func processEvent(_ event: ConversationAccessUpdateEvent) async throws
+    func processEvent(_ event: ConversationAccessUpdateEvent) async
 
 }
 
 struct ConversationAccessUpdateEventProcessor: ConversationAccessUpdateEventProcessorProtocol {
 
-    func processEvent(_: ConversationAccessUpdateEvent) async throws {
-        // TODO: [WPB-10164]
-        assertionFailure("not implemented yet")
+    let context: NSManagedObjectContext
+    let repository: any ConversationRepositoryProtocol
+
+    func processEvent(_ event: ConversationAccessUpdateEvent) async {
+        let conversationID = event.conversationID
+
+        let localConversation = await repository.fetchOrCreateConversation(
+            with: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let accessRoles = if let legacyAccessRole = event.legacyAccessRole {
```

Comment:
Hmm... do we even need to handle the legacy roles anymore? The newer roles are always present.

Reply:
it's actually an optional at the event decoder level, we default it to empty set if this is nil. in versioned API `access_role_v2`  is removed in v3 and renamed to `access_role` but since we're dealing with an event here (not versioned then) we assume we receive either one of these two properties. I did some changes to be more explicit in https://github.com/wireapp/wire-ios/pull/2028/commits/75042b6e13efa52d32e7103943e04a74c65a1af6 and https://github.com/wireapp/wire-ios/pull/2028/commits/d00fd5f66021960357311b2b55337b538169fb94

Reply:
ok thanks for clarifying. Perhaps a short comment is needed here